core/src/main/java/org/jboss/logmanager/LogContext.java

```diff
@@ -142,7 +164,14 @@ public static LogContext create() {
      */
     @SuppressWarnings({ "unchecked" })
     public <V> V getAttachment(Logger.AttachmentKey<V> key) {
-        return rootLogger.getAttachment(key);
+        if (key == null) {
+            throw new NullPointerException("key is null");
```

Comment:
We could use `wildfly-common` for these `null` checks as it's already a required dependency. Doesn't matter much either way to me though.